README.md

```diff
@@ -273,7 +282,107 @@ should mention how to instantiate those variables/files, etc.
+It is a testing method by which individual units of source code are tested to determine if they are ready to use. 
+It helps to reduce the cost of bug fixes since the bugs are identified during the early phases of the development lifecycle.
+
+What is Integration Test?
+Integration testing tests integration between software modules. It is a software testing technique where individual units of a program 
+are combined and tested as a group. It checks the overall flow of the application after the integration of different modules.
+
+
+
+#### Test driven development
+Test-driven development (TDD) is a software development process that relies on the repetition of a very short development cycle: requirements are turned into 
+very specific test cases, then the code is improved so that the tests pass. In simple terms, 
+test cases for each functionality are created and tested first and if the test fails then the new code is written in order to pass the test 
+and making code simple and bug-free.
+
+
+Useful links for writing tests in our most common  languages and frameworks: 
+* [Python](https://realpython.com/python-testing/)
+* [React](https://reactjs.org/docs/testing-recipes.html)
+* [Javascript](https://jestjs.io/)
+* [R](https://r-pkgs.org/tests.html)
+
+## Continuous integration and delivery
+
+The CI/CD pipeline is one of the best practices for devops teams to implement, 
+for delivering code changes more frequently and reliably. It is also an agile methodology 
+best practice, as it enables software development teams to focus on meeting business requirements, 
+and code quality because deployment steps are automated.
+
+### Continuous integration
+Developers practicing continuous integration merge their changes back to the main branch as often as possible. 
+The developer's changes are validated by creating a build and running automated tests against the build. 
+By doing so, you avoid the integration hell that usually happens when people wait for release day to merge
+ their changes into the release branch.
+
+Continuous integration puts a great emphasis on testing automation to check that the application is not broken 
+whenever new commits are integrated into the main branch.
+
+#### What you need (cost)
+Your team will need to write automated tests for each new feature, improvement or bug fix. It should also  
+limit the size of the changes to make them easier to review and merge as soon as they're ready.
+You need a continuous integration server that can monitor the main repository and run the tests automatically. 
+
+#### What you gain
+Less bugs get shipped to production as regressions are captured early by the automated tests.
+Building the release is easy as all integration issues have been solved early.
+Less context switching as developers are alerted as soon as they break the build and can work on fixing it before they move to another task.
```

Comment:
I associate alerts with more context switching.

Reply:
I used this practices in the past and found them useful. But this is just my personal experience.

Reply:
I also find them useful! That was not the point here. It was that this particular sentence seems to suggest that these alerts leads to less context switching. The important thing here is that this is a practice that might lead to that the build is broken less.